### PR TITLE
Switch to docker(-py) >= 2

### PR DESCRIPTION
**Pull request by fbochu**

### CODE REVIEW

File: src/grocker/builders.py [hunk 1]
```diff
@@ -260,13 +251,15 @@ def docker_get_client(**kwargs):
...
     all_extra_kwargs['version'] = DOCKER_API_VERSION
-    client = docker.Client(**all_extra_kwargs)
+    client = docker.DockerClient(**all_extra_kwargs)
     client.version()  # Call the API, this will raise if API version is not compatible.
     return client
 
 
 def docker_build_image(docker_client, path, tag=None, pull=True, buildargs=None):
-    stream = docker_client.build(path=path, tag=tag, rm=True, forcerm=True, pull=pull, buildargs=buildargs)
+    stream = docker_client.images.client.api.build(
```
Comment (xavfernandez) [inline src/grocker/builders.py]: `docker_client.api.build` should be enough

Comment (xavfernandez) [inline src/grocker/builders.py]: You could also add `decode=True` to auto-magically get decoded dicts and avoid the ugly `docker.utils.json_stream.json_stream` in `inspect_stream`

File: src/grocker/builders.py [hunk 2]
```diff
@@ -304,23 +296,25 @@ def inspect_stream(stream):
 
 def docker_pull_image(docker_client, name):
     logger.info('Pulling %s...', name)
-
-    stream = docker_client.pull(name, stream=True)
-    inspect_stream(stream)
-    return [image for image in docker_client.images() if name in (image['RepoTags'] or [])]
+    try:
+        stream = docker_client.images.pull(name, stream=True)
```
Comment (xavfernandez) [inline src/grocker/builders.py]: I think we could use the low-level API here, with `docker_client.api.pull` (and `decode=True`) and provide a nice `--verbose` option to decide what logs we want to show.

File: src/grocker/builders.py [hunk 3]
```diff
@@ -304,23 +296,25 @@ def inspect_stream(stream):
...
+        return []
+    return [image for image in docker_client.images.list() if name in (image['RepoTags'] or [])]
 
 
 def docker_push_image(docker_client, name):
     logger.info('Pushing %s...', name)
 
-    stream = docker_client.push(name, stream=True)
+    stream = docker_client.images.push(name, stream=True)
     data = inspect_stream(stream)
```
Comment (xavfernandez) [inline src/grocker/builders.py]: Same as for `docker_pull_image` (and `docker_build_image`)

File: src/grocker/resources/docker/runner-image/provision.sh [hunk 4]
```diff
@@ -51,7 +51,7 @@ only_run_as_root() {  # script_or_function
 
 
 provision() {
-    setup_venv ~/app.venv ${GROCKER_RUNTIME} ${GROCKER_APP}[${GROCKER_APP_EXTRAS}]==${GROCKER_APP_VERSION}
+    setup_venv ~/app.venv ${GROCKER_RUNTIME:=python3} "${GROCKER_APP:=}[${GROCKER_APP_EXTRAS:=}]==${GROCKER_APP_VERSION:=}"
```
Comment (xavfernandez) [inline src/grocker/resources/docker/runner-image/provision.sh]: Not sure these default values are needed/a good idea.
We want those environment variables to exist.

File: tests/test_config.py [hunk 5]
```diff
@@ -11,8 +11,8 @@
 import itertools
 
 from grocker import __version__
-from grocker.__main__ import parse_config
-from grocker.helpers import default_image_name
+from grocker.utils import parse_config
+from grocker.utils import default_image_name
```
Comment (xavfernandez) [inline tests/test_config.py]: Shouldn't this be part of `Split helpers into helpers and utils` commit ?

File: src/grocker/builders/naming.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+# Copyright (c) Polyconseil SAS. All rights reserved.
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from .. import __version__
+from .. import utils
+
+logger = logging.getLogger(__name__)
```
Comment (xavfernandez) [inline src/grocker/builders/naming.py]: Not used

File: tests/test_config.py (hunk 5 above)
Comment (fbochu) [inline tests/test_config.py]: All `-w-` commits will be merge with no `-w-` commit before pushing on master branch

**Comment by fbochu** [comment]
Replaced by : 
- https://github.com/Polyconseil/grocker/pull/133
- https://github.com/Polyconseil/grocker/pull/134
- https://github.com/Polyconseil/grocker/pull/135

**Pull request closed without merging**